File: Data Structures & Algorithms/binary-search/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let l = 0;
        let r = nums.length - 1;
        while (l <= r) {
            let middle = Math.floor(l + (r - l) / 2);
            if (nums[middle] == target) return middle;
            if (nums[middle] < target) {
                l = middle + 1;
            } else {
                r = middle - 1;
            }
        }

        return -1;

    }
}

File: Data Structures & Algorithms/sort-colors/submission-2.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int l = 0, m = 0, r = 0;
        for (int i = 0; i < nums.size(); ++i) {
            // [2, 1, 0]
            // [lm, r]
            // [1 , 2, 0]
            // [l, mr]
            // [0, 2, 1]
            // [, lmr]
            if (nums[i] == 0) {
                swap(nums[l], nums[i]);
                if (nums[i] == 1) {
                    swap(nums[i], nums[m]);
                }
                l++;
                m++;
                r++;
            } else if (nums[i] == 1) {
                swap(nums[m], nums[i]);
                m++;
                r++;
            } else {
                swap(nums[r], nums[i]);
                r++;
            }
        }
    }
};